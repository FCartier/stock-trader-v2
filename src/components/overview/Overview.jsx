import React from "react";
import { SectionHeader } from "../reusables/SectionHeader"

const Overview = ({overview}) => {
  return Object.keys(overview).length ? (
    <div>
        <div className="overview">
          <SectionHeader title="COMPANY OVERVIEW"/>
          <div className="overview-content">
            <h2>{overview.companyName}</h2>
            <p>{overview.website}</p>
            <p>{overview.description}</p>
          </div>
        </div>
    </div>
  ) : null
};

export default Overview;
