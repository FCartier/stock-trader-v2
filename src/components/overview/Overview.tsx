import * as React from "react";
import { SectionHeader } from "../reusables/SectionHeader"

export interface IOverviewProps {
  overview?: {
      companyName: string;
      website: string;
      description: string;
  };
}

const Overview: React.SFC<IOverviewProps> = ({overview}) => {
  return !(typeof overview === "undefined" || overview.companyName === "") ? (
    <div>
      <div className='overview'>
        <SectionHeader title="COMPANY OVERVIEW" />
        <div className='overview-content'>
          <h2>{overview.companyName}</h2>
          <p>{overview.website}</p>
          <p>{overview.description}</p>
        </div>
      </div>
    </div>
  ) : null;
};

export default Overview;
