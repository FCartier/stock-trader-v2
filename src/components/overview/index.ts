import OverviewContainer from "./overviewContainer";

export default OverviewContainer;