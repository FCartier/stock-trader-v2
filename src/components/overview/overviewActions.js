export const OVERVIEW_ACTION = "OVERVIEW ACTION";

export const overviewAction = payload => ({
    type: OVERVIEW_ACTION,
    payload
  });