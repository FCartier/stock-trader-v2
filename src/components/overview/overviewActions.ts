export const FETCH_OVERVIEW_SUCCESS = "FETCH_OVERVIEW_SUCCESS";
export type FETCH_OVERVIEW_SUCCESS = typeof FETCH_OVERVIEW_SUCCESS;

export const FETCH_OVERVIEW_FAILED = "FETCH_OVERVIEW_FAILED";
export type FETCH_OVERVIEW_FAILED = typeof FETCH_OVERVIEW_FAILED;