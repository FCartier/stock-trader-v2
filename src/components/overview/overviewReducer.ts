import { Reducer } from "redux";
import {
  FETCH_OVERVIEW_SUCCESS,
  FETCH_OVERVIEW_FAILED
} from "./overviewActions";
import OverviewState from "./types";
import { fetchStatus } from "../reusables/fetchStatus";

const initialState: OverviewState = {
  overview: {
    companyName: "",
    website: "",
    description: ""
  },
  status: fetchStatus.pending
};

export const overview: Reducer<OverviewState> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case FETCH_OVERVIEW_SUCCESS:
      return { overview: action.payload, status: fetchStatus.success };
    case FETCH_OVERVIEW_FAILED:
      return { ...state, status: fetchStatus.failed };
    default:
      return state;
  }
};
