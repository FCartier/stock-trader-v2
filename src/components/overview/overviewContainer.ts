import Overview from "./Overview";
import { connect } from "react-redux";
import { GlobalState } from "../../StoreTypes";

export const mapStateToProps = ({ overview: { overview, status } }: GlobalState) => ({
  overview,
  status
});

export default connect(
  mapStateToProps,
  null
)(Overview);
