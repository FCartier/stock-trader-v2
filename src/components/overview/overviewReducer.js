import { OVERVIEW_ACTION } from "./overviewActions"

export const overview = (state = {}, action) => {
    switch (action.type) {
      case OVERVIEW_ACTION:
        return action.payload;
      default:
        return state;
    }
  };