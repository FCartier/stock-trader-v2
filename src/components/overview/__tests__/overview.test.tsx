import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import Overview from "../Overview";
import { testData } from "../__data__/overview.data";
import { fetchStatus } from "../../reusables/fetchStatus";

configure({ adapter: new Adapter() });

describe("overview component", () => {
  test("overview component should render correctly before fetch", () => {
    const component = shallow(
      <Overview
        overview={{
          companyName: "",
          website: "",
          description: ""
        }}
        status={fetchStatus.pending}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("overview component should render correctly with fetch success", () => {
    const component = shallow(
      <Overview overview={testData} status={fetchStatus.success} />
    );
    expect(component).toMatchSnapshot();
  });

  test("overview component should render correctly with fetch failed", () => {
    const component = shallow(
      <Overview overview={testData} status={fetchStatus.failed} />
    );
    expect(component).toMatchSnapshot();
  });
});
