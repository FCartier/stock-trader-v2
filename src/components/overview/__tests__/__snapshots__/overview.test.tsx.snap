// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview component overview component should render correctly before fetch 1`] = `""`;

exports[`overview component overview component should render correctly with fetch failed 1`] = `
<div>
  <div
    className="overview"
  >
    <Component
      title="COMPANY OVERVIEW"
    />
    <div
      className="overview-content"
    >
      <NoData
        componentName="overview"
      />
    </div>
  </div>
</div>
`;

exports[`overview component overview component should render correctly with fetch success 1`] = `
<div>
  <div
    className="overview"
  >
    <Component
      title="COMPANY OVERVIEW"
    />
    <div
      className="overview-content"
    >
      <h2>
        American Assets Trust Inc.
      </h2>
      <p>
        http://www.americanassetstrust.com
      </p>
      <p>
        American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
      </p>
    </div>
  </div>
</div>
`;
